extensions/ql-vscode/src/query-history.ts

```diff
@@ -611,7 +613,27 @@ export class QueryHistoryManager extends DisposableObject {
       await this.refreshTreeView();
     });
 
+    const variantAnalysisStatusUpdateSubscription = this.variantAnalysisManager.onVariantAnalysisStatusUpdated(async (variantAnalysis) => {
+      const item = this.treeDataProvider.allHistory.find(i => i.t === 'variant-analysis' && i.historyItemId === variantAnalysis.id.toString());
```

Comment:
Sorry for missing this earlier, but it seems like the `historyItemId` is actually a `nanoid`, so this would never match any history items. Instead, I think we'd need to check `i.variantAnalysis.id`.

Reply:
Hmmm, I was under the impression that we decided to go with a `nanoid` specifically to be consistent with the way we identify all the other types of query history items: https://github.com/github/vscode-codeql/pull/1614#discussion_r996891436

Reply:
I think that's correct, but that does mean we need to do something like this instead:

```suggestion
      const item = this.treeDataProvider.allHistory.find(i => i.t === 'variant-analysis' && i.variantAnalysis.id === variantAnalysis.id);
```

Essentially, the `historyItemId` is completely independent of the variant analysis and cannot be used to find the matching variant analysis. That also means that we don't have a 1-to-1 relationship between a variant analysis and a query history item. It might even be the case that there can be multiple history items for the same variant analysis (at least in theory), so it might even make sense to make this a `filter` call and update all matching history items.

Reply:
Alrighty, I've changed this.

Reply:
Have also changed this to use `filter`: https://github.com/github/vscode-codeql/pull/1620/commits/3a7f2000622e65cacf489a90515c57c9a923df2a

Reply:
Just a note that it's not out of the picture to change the `historyItemId` to be something determined from the variant analysis ID instead of a random ID. The thinking about a "1-to-1 relationship between a variant analysis and a query history item" does make that seem a bit more attractive. But I don't see it as hugely urgent so we can leave things as they are for now and in the next couple of weeks see if this choice would make things better overall. How does that sound?

Reply:
Sounds good, I've written up an issue for this as well! 

:all-the-issues-emoji: